simtools/corsika/corsika_config.py

```diff
@@ -0,0 +1,522 @@
+import logging
+import random
+
+import astropy.units as u
+from astropy.io.misc import yaml
+
+import simtools.config as cfg
+import simtools.io_handler as io
+import simtools.util.general as gen
+from simtools.util import names
+from simtools.layout.layout_array import LayoutArray
+from simtools.util.general import collectDataFromYamlOrDict
+
+__all__ = [
+    'CorsikaConfig',
+    'MissingRequiredInputInCorsikaConfigData',
+    'InvalidCorsikaInput'
+]
+
+
+class MissingRequiredInputInCorsikaConfigData(Exception):
+    pass
+
+
+class InvalidCorsikaInput(Exception):
+    pass
+
+
+class CorsikaConfig:
+    '''
+    CorsikaConfig deals with configuration for running CORSIKA. \
+    User parameters must be given by the corsikaConfigData or \
+    corsikaConfigFile arguments. An example of corsikaConfigData follows \
+    below.
+
+    .. code-block:: python
+
+    corsikaConfigData = {
+        'primary': 'proton',
+        'nshow': 10000,
+        'nrun': 1,
+        'zenith': 20 * u.deg,
+        'viewcone': 5 * u.deg,
+        'erange': [10 * u.GeV, 100 * u.TeV],
+        'eslope': -2,
+        'phi': 0 * u.deg,
+        'cscat': [10, 1500 * u.m, 0]
+    }
+
+    The remaining CORSIKA parameters can be set as a yaml file, using the argument \
```

Comment:
Discuss later again the difference between 'user parameters' and 'corsika parameters'. It seems the division is sometimes not entirely clear?

Reply:
I will add this as a topic for our call.

Reply:
(ok - still need to discuss this but should not prevent us from merging here)